datavec/datavec-python/src/main/java/org/datavec/python/PythonTransform.java

```diff
@@ -0,0 +1,261 @@
+package org.datavec.python;
+
+import org.datavec.api.transform.ColumnType;
+import org.datavec.api.transform.Transform;
+import org.datavec.api.transform.schema.Schema;
+import org.datavec.api.writable.*;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import javax.annotation.Nullable;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Row-wise Transform that applies arbitrary python code on each row
+ */
+public class PythonTransform implements Transform{
+    private String setupCode;
+    private String execCode;
+    private String code;
+    private PythonVariables pyInputs;
+    private PythonVariables pyOutputs;
+    private String name;
+    private Schema inputSchema;
+    private Schema outputSchema;
+
+
+    public PythonTransform(String code, PythonVariables pyInputs, PythonVariables pyOutputs) throws Exception{
+        this.code = code;
+        this.pyInputs = pyInputs;
+        this.pyOutputs = pyOutputs;
+        parseSetupAndExecCode();
+        this.name = UUID.randomUUID().toString();
+    }
+
+    @Override
+    public void setInputSchema(Schema inputSchema){
+        this.inputSchema = inputSchema;
+        try{
+            pyInputs = schemaToPythonVariables(inputSchema);
+        }catch (Exception e){
+            throw new RuntimeException(e);
+        }
+        if (outputSchema == null){
+            outputSchema = inputSchema;
+        }
+
+    }
+
+    @Override
+    public Schema getInputSchema(){
+        return inputSchema;
+    }
+
+    @Override
+    public List<List<Writable>> mapSequence(List<List<Writable>> sequence) {
+        List<List<Writable>> out = new ArrayList<>();
+        for (List<Writable> l : sequence) {
+            out.add(map(l));
+        }
+        return out;
+    }
+
+    @Override
+    public Object map(Object input) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public Object mapSequence(Object sequence) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public List<Writable> map(List<Writable> writables){
+        PythonVariables pyInputs = getPyInputsFromWritables(writables);
+        try{
+            PythonVariables pyOutputs = PythonExecutioner.getInstance().exec(this, pyInputs);
+            return getWritablesFromPyOutputs(pyOutputs);
+        }
+        catch (Exception e){
+            throw new RuntimeException(e);
```

Comment:
@AlexDBlack is this normal for transforms? Have we changed anything in the data vec transforms you can think of?